### PR TITLE
gds/ds12: plug a memory leak in pmix_ds12_lock_finalize()

### PR DESCRIPTION
**Pull request by ggouaillardet**
Do release the lockfile.

Signed-off-by: Gilles Gouaillardet <gilles@rist.or.jp>

### CODE REVIEW

**Comment by ggouaillardet** [comment]
Refs. pmix/prrte#156

File: src/mca/gds/ds12/gds_ds12_lock_pthread.c [hunk 1]
```diff
@@ -216,6 +216,7 @@ void pmix_ds12_lock_finalize(pmix_common_dstor_lock_ctx_t *lock_ctx)
     pmix_pshmem.segment_detach(pthread_lock->segment);
 
     free(pthread_lock->segment);
+    free(pthread_lock->lockfile);
```
Comment (rhc54) [inline src/mca/gds/ds12/gds_ds12_lock_pthread.c]: Probably should set this to NULL after free?

**Comment by karasevb** [comment]
But I would suggest checking them separately:
```C
    if (NULL == pthread_lock->lockfile) {
        PMIX_ERROR_LOG(PMIX_ERROR);
        return;
    }
    if (NULL == pthread_lock->segment) {
        PMIX_ERROR_LOG(PMIX_ERROR);
        return;
    }
```

**Comment by ggouaillardet** [comment]
@rhc54 @karasevb I made the suggested changes.

**Comment by karasevb** [comment]
@ggouaillardet Thanks!

**Pull request merged**